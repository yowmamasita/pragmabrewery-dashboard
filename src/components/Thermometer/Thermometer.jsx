import React from 'react';

export const Thermometer = ({lowerBound, upperBound, lowestSafeTemp, highestSafeTemp, currentTemp}) =>
  {
    const range = Math.abs(upperBound - lowerBound);
    const maxWidth = window.innerWidth;
    const unitWidth = maxWidth / range;
    const lowUnsafeWidth = Math.abs(lowestSafeTemp-lowerBound) * unitWidth;
    const highUnsafeWidth = Math.abs(upperBound-highestSafeTemp) * unitWidth;
    const safeWidth = Math.abs(highestSafeTemp-lowestSafeTemp) * unitWidth;
    const currentTempX = (currentTemp - lowerBound) * unitWidth;
    console.log(unitWidth);
    console.log(lowUnsafeWidth, lowestSafeTemp, lowerBound);
    console.log(highUnsafeWidth, upperBound, highestSafeTemp);
    console.log(safeWidth, highestSafeTemp, lowestSafeTemp);
    return (
      <svg width={maxWidth} height="50" version="1.1" xmlns="http://www.w3.org/2000/svg">
        <rect x={0} y={10} width={lowUnsafeWidth} height={30} stroke="#3498DB" fill="#3498DB" stroke-width="1"/>
        <rect x={lowUnsafeWidth} y={10} width={safeWidth} height={30} stroke="#52BE80" fill="#52BE80" stroke-width="1"/>
        <rect x={lowUnsafeWidth+safeWidth} y={10} width={highUnsafeWidth} height={30} stroke="#E74C3C" fill="#E74C3C" stroke-width="1"/>
        <rect x={currentTempX} y={5} width={5} height={40} stroke="black" fill="black" stroke-width="1"/>
      </svg>
    );
  }
