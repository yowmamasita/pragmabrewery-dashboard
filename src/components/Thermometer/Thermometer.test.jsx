import React from 'react';
import renderer from 'react-test-renderer';
import {Thermometer} from './Thermometer';

it('renders correctly', () => {
  const thermometer = renderer
    .create(<Thermometer lowerBound={-2} upperBound={10} lowestSafeTemp={5} highestSafeTemp={7} currentTemp={5.5}/>)
    .toJSON();
  expect(thermometer).toMatchSnapshot();
});
