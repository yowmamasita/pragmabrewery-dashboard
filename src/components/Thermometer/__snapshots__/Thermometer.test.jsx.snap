// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<svg
  height="50"
  version="1.1"
  width={600}
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="#3498DB"
    height={30}
    stroke="#3498DB"
    stroke-width="1"
    width={350}
    x={0}
    y={10}
  />
  <rect
    fill="#52BE80"
    height={30}
    stroke="#52BE80"
    stroke-width="1"
    width={100}
    x={350}
    y={10}
  />
  <rect
    fill="#E74C3C"
    height={30}
    stroke="#E74C3C"
    stroke-width="1"
    width={150}
    x={450}
    y={10}
  />
  <rect
    fill="black"
    height={40}
    stroke="black"
    stroke-width="1"
    width={5}
    x={375}
    y={5}
  />
</svg>
`;
